### PR TITLE
Include libsoup-2.4

### PR DESCRIPTION
**Pull request by sergio-costas**
Since libsoup-2.4 has been removed from gnome-46-2404, it must be included directly in the snap, along with librest-0.7.

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Built with the candidate version of gnome-46-2404-sdk and ran with the candidate version of gnome-46-2404.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by sergio-costas** [comment]
It fails to build because it requires https://github.com/ubuntu/gnome-sdk/issues/260

File: snapcraft.yaml [hunk 1]
```diff
@@ -25,6 +25,9 @@ description: |
 grade: stable
 confinement: strict # use 'strict' once you have the right plugs and slots
 
+environment:
+  LD_LIBRARY_PATH: $SNAP/usr/lib64:$SNAP/usr/lib:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
```
Comment (seb128) [inline snapcraft.yaml]: Why is that needed now? Would be nice to add a comment explaining why that's not handled by the desktop launcher

File: snapcraft.yaml [hunk 2]
```diff
@@ -78,7 +121,6 @@ parts:
     build-snaps:
       - gnome-46-2404-sdk/latest/candidate
 
-
```
Comment (seb128) [inline snapcraft.yaml:81]: Cleanups are ok but that doesn't have to do with the commit description, could you split it to a separate commit?

File: snapcraft.yaml [hunk 3]
```diff
@@ -93,9 +135,9 @@ parts:
       for snap in "core24" "gnome-46-2404"; do
         cd "/snap/$snap/current/usr/lib"
         for filename in [ *.so* ]; do
-          rm -f "$CRAFT_PRIME/usr/lib/$CRAFT_ARCH_TRIPLET/$filename"
+          rm -f "$CRAFT_PRIME/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/$filename"
         done
-        cd "/snap/$snap/current/usr/lib/$CRAFT_ARCH_TRIPLET"
+        cd "/snap/$snap/current/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR"
```
Comment (seb128) [inline snapcraft.yaml]: Similarly, those changes are for core24 deprecation right? They make sense but aren't part of including libsoup so should be split in another commit

**Review by seb128 — CHANGES_REQUESTED** [review]
Thanks. I'm fine with building it but it makes the snap build longer, did you consider staging the libsoup deb instead? Or any reason that wouldn't work?

I've also some inline nitpicks

File: snapcraft.yaml (hunk 1 above)
Comment (sergio-costas) [reply snapcraft.yaml]: It was needed because, by default, autotools plugin install libraries at /usr/lib64 instead of /usr/lib/x86_64-linux-gnu. I fixed that, so it's not needed anymore.

File: snapcraft.yaml (hunk 2 above)
Comment (sergio-costas) [reply snapcraft.yaml:81]: Removed.

File: snapcraft.yaml (hunk 3 above)
Comment (sergio-costas) [reply snapcraft.yaml]: Removed. Will send another PR.

Comment (sergio-costas) [reply snapcraft.yaml]: Also, used libsoup and libgoa packages, as requested.

File: snapcraft.yaml [hunk 4]
```diff
@@ -73,16 +74,23 @@ parts:
...
+      - libgoa-1.0-0b
     build-snaps:
       - gnome-46-2404-sdk/latest/candidate
 
-
   cleanup:
     after: [ gnome-recipes ]
     plugin: nil
-    build-snaps: [core24, gtk-common-themes, gnome-46-2404]
+    build-snaps: [core24, gtk-common-themes, gnome-46-2404/latest/candidate]
```
Comment (seb128) [inline snapcraft.yaml:93]: I don't think we should hardcode the candidate there. Is there anything that is going to break if we build again stable of just the cleanup isn't going to be optimal? If there is a breakage we should imho just wait for gnome-46-2404 candidate to promote to stable, otherwise we can merge already and we will do another rebuild after promotion...

**Review by seb128 — CHANGES_REQUESTED** [review]

File: snapcraft.yaml (hunk 4 above)
Comment (sergio-costas) [reply snapcraft.yaml:93]: The problem is that, until the current candidate goes into stable, it will take the current gnome-46-2404 stable, where libsoup-2.4 still exists, and remove it from the snap, so it won't work. So either we publish this now and remove that "/latest/candidate" part when the current gnome-46-2404 candidate goes stable, or we remove that now and wait until gnome-46-2404 candidate goes stable to publish it.

Comment (seb128) [reply snapcraft.yaml:93]: alright

**Review by seb128 — APPROVED** [review]
let's land this one, we will revert the /candidate later
